### ISSUE TITLE
Duplicate messages with corrupt data using `-gbrk`

### ISSUE THREAD

**Issue by electrodude** [totalspectrum/spin2cpp#274]
The following code, when compiled with `-gbrk`, prints out `message 1` twice, and nothing else.  It should print `message 1` exactly once.
```spin
PUB main | x

  debug("message 1")

  ifnot @x
    debug("message 2")

  repeat
```

Here's a similar example, this time with formatted variables.  The data in the second message is bogus.
```spin
PUB main | x, y

  debug("data ", uhex_long(@x))

  ifnot @main
    ' try either of these
    'debug("message 2")
    debug("DATA ", uhex_long(@y))

  repeat
```
It prints this:
```

**Comment by electrodude** [comment]
I'm pretty sure the thing with the corrupt data in the duplicate `debug` message (as opposed to duplicated messages having the same data) happened before I started using any method pointers, but I couldn't find a simple example without method pointers just now.

In the project I'm working on, I'll often get more than one duplicate of the same message, each with different garbage data.  For example, one run of this code seems to have produced all three of the following messages:
```spin
  ' (an if and some elseifs)
  else
    debug("rx UDP ", uhex_word(rx_len), ": ", udec_(sport), " => ", udec_(dport), ": unhandled")
    abort 0
```
```
Cog0  rx UDP rx_len = $012C: 68 => 67: unhandled
Cog0  rx UDP rx_len = $0910: 4_052_102_660 => 4_052_103_172: unhandled
Cog0  rx UDP rx_len = $051F: 4_234_674_464 => 4_043_718_660: unhandled
```
The first message is correct, but `4_052_102_660` is not a valid UDP port number, and code before the snippet I gave guarantees that `sport` and `dport` can never exceed 16 bits anyway.

Normally it's obvious which messages are bogus, but if this happens to a message containing e.g. `uhex_byte_array(ptr, len)` and the length field ends up finding an absurdly large value, the P2 hangs while it busily floods the terminal with an endless stream of data.

**Comment by totalspectrum** [comment]
Interesting... these always seem to be conditional (inside an IF statement) and the BRKs have been conditionalized. I've checked in a change to never conditionalize BRK (to always jump over it instead). If this fixes your problem (it does fix the first simple example) then it may indicate a potential hardware bug...

**Comment by Wuerfel21** [comment]
Yep, BRK can't be conditional. That's because the BRK instruction itself only sets the break code. The actual soft interrupt is triggered by the presence of the BRK opcode in the pipeline (and at the point that this is checked, the previous instruction hasn't completed so checking the flags would be bogus, anyways). Chip explained this at length on Zoom once, idk if its been recorded.

**Comment by electrodude** [comment]
This fixed my problems.  Thanks!
